go/vt/vtgate/engine/merge_sort.go

```diff
@@ -27,24 +25,61 @@ import (
 	"vitess.io/vitess/go/sqltypes"
 
 	querypb "vitess.io/vitess/go/vt/proto/query"
-	"vitess.io/vitess/go/vt/srvtopo"
+	vtrpcpb "vitess.io/vitess/go/vt/proto/vtrpc"
+	"vitess.io/vitess/go/vt/vterrors"
 )
 
-// MergeSort performs a merge-sort of rows returned by a streaming scatter query.
-// Each shard of the scatter query is treated as a stream. One row from each stream
-// is added to the merge-sorter heap. Every time a value is pulled out of the heap,
+// StreamExecuter is a subset of Primitive that MergeSort
+// requires its inputs to satisfy.
+type StreamExecuter interface {
+	StreamExecute(vcursor VCursor, bindVars map[string]*querypb.BindVariable, wantields bool, callback func(*sqltypes.Result) error) error
+}
+
+var _ Primitive = (*MergeSort)(nil)
+
+// MergeSort performs a merge-sort of rows returned by each Input. This should
+// only be used for StreamExecute. One row from each stream is added to the
+// merge-sorter heap. Every time a value is pulled out of the heap,
 // a new value is added to it from the stream that was the source of the value that
 // was pulled out. Since the input streams are sorted the same way that the heap is
 // sorted, this guarantees that the merged stream will also be sorted the same way.
-func MergeSort(vcursor VCursor, query string, orderBy []OrderbyParams, rss []*srvtopo.ResolvedShard, bvs []map[string]*querypb.BindVariable, callback func(*sqltypes.Result) error) error {
+// MergeSort is not functionally complete and should not be used as a Primitive
```

Comment:
Can you add a comment here explaining why it still needs to implement the Primitive interface?
IOW, how/where it is being used as a Primitive.